/**:
  ros__parameters:
    log_dir: /home/yuchen/ASRL/temp/doppler_odometry/boreas/aeva/steam

    # SLAM OPTIONS ----
    max_num_threads: 1
    save_trajectory: true
    suspend_on_failure: false
    output_dir: /home/yuchen/ASRL/temp/doppler_odometry/boreas/aeva/steam
    all_sequences: true
    sequence: "" # Ignored if use_all_sequences is true
    start_index: 0 # The initial frame index for the given sequence
    max_frames: -1 # For negative values, all frames are computed
    with_viz3d: false
    viz_mode: AGGREGATED

    # VISUALIZATION OPTIONS ----
    visualization_options:
      odometry: true
      raw_points: true
      sampled_points: true
      map_points: true
      T_sr_vec: [-0.835, -0.39, 0.103, 0.0, 0.0, 0.0] # davids result without side slip

    # DATASET OPTIONS ----
    dataset_options:
      dataset: AEVA # Options: [KITTI_raw / KITTI_CARLA / KITTI / KITTI-360 / BOREAS / AEVA]
      root_path: /home/yuchen/ASRL/data/boreas/sequences # Set to the root path of kitti's directory
      fail_if_incomplete: false
      min_dist_lidar_center: 5.0 # Suppresses points to close to the center of the sensor
      max_dist_lidar_center: 100.0 # Suppresses points to far from the center of the sensor

    # ODOMETRY OPTIONS ----
    odometry_options:
      init_num_frames: 20
      voxel_size: 0.5 # The voxel size for the grid sampling of the new frame (before keypoints extraction)
      sample_voxel_size: 1.5 # The size of a voxel for the selection of `keypoints` by grid sampling
      max_distance: 100.0 # The threshold of the distance to suppress voxels from the map
      max_num_points_in_voxel: 20 # The maximum number of points per voxel of the map
      min_distance_points: 0.1
      distance_error_threshold: 5.0 # The motion of the sensor between two frames which is considered erroneous (stops the odometry)
      robust_registration: false
      motion_compensation: CONTINUOUS # The profile of the motion compensation (NONE, CONSTANT_VELOCITY, ITERATIVE, CONTINUOUS)
      initialization: INIT_CONSTANT_VELOCITY
      debug_print: true

      # CT_ICP OPTIONS ----
      ct_icp_options:
        size_voxel_map: 1.0 # The voxel size of in the voxel map
        num_iters_icp: 10 # The number of iterations of the ICP
        min_number_neighbors: 20 # The minimum number of neighbor points to define a valid neighborhood
        voxel_neighborhood: 1
        max_number_neighbors: 20
        max_dist_to_plane_ct_icp: 0.3
        point_to_plane_with_distortion: true
        distance: CT_POINT_TO_PLANE # Options: [CT_POINT_TO_PLANE, POINT_TO_PLANE]
        num_closest_neighbors: 1
        beta_location_consistency: 0.001
        beta_constant_velocity: 0.001

        solver: STEAM # Options: [GN=Gauss-Newton, CERES, STEAM]
        loss_function: CAUCHY # Options: [STANDARD, CAUCHY, HUBER, TOLERANT, TRUNCATED] (for CERES solver)
        ls_max_num_iters: 5 # The number of steps performed for each iteration of the ICP (for CERES solver)
        ls_num_threads: 10 # The number of threads to build and solve the least square system (for CERES solver)
        ls_sigma: 0.1 # The sigma parameter for loss CAUCHY, HUBER, TOLERANT, TRUNCATED (for CERES solver)
        ls_tolerant_min_threshold: 0.05 # The tolerant parameter for loss TOLERANT (for CERES solver)

        debug_print: true

        steam:
          T_sr_vec: [-0.835, -0.39, 0.103, 0.0, 0.0, 0.0] # davids result without side slip

          qc_inv_diag: [0.05, 0.05, 0.05, 0.05, 0.05, 0.05]
          num_extra_states: 0
          add_prev_state: true
          lock_prev_pose: false
          lock_prev_vel: false
          prev_pose_as_prior: true
          prev_vel_as_prior: true
          no_prev_state_iters: 10
          association_after_adding_prev_state: false

          use_vp: false
          vp_cov_diag: [1000.0, 0.001, 0.001, 1.0, 1.0, 1000.0]

          use_rv: false
          merge_p2p_rv: false
          rv_cov_inv: 0.1
          rv_loss_threshold: 0.1

          verbose: false
          max_iterations: 1
          num_threads: 10
